src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,98 @@
+    }
+
+    /**
+     * Indicate to the user that their command has been successfully executed or if there is any error along the way.
+     */
+    public void allTasks(String inputText, TaskList tasks) {
+        int taskCount = tasks.list.size();
+        List<Task> list = tasks.list;
+        String[] tempList = inputText.split(" ", 2);
+
+        try {
+            new DukeException().invalidChecker(tempList, taskCount);
+        } catch (DukeException e) {
+            System.err.println(e);
+            return;
+        }
+
+        System.out.println("------------------------------------------------------------");
+        switch (tempList[0]) {
+            case "bye":
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            case "list":
+                System.out.println("Here are the tasks in your list:");
+                ShowTasks.showList(taskCount, list);
+                break;
+            case "find":
+                System.out.println("Here are the matching tasks in your list:");
+                ShowTasks.find(taskCount, tempList[1], list);
+                break;
+            case "todo":
+                CreateTask.todo(tempList[1], list);
+                System.out.println(list.get(taskCount).toString());
+                System.out.println("Got it. I've added this task:");
+                printAndSave(taskCount + 1);
+                break;
+            case "deadline":
+                CreateTask.deadline(tempList[1], list);
+                System.out.println(list.get(taskCount).toString());
+                printAndSave(taskCount + 1);
+                System.out.println("Got it. I've added this task:");
+                break;
+            case "event":
+                CreateTask.event(tempList[1], list);
+                System.out.println(list.get(taskCount).toString());
+                printAndSave(taskCount + 1);
+                System.out.println("Got it. I've added this task:");
+                break;
+            case "mark":
+                MarkTask.mark(tempList[1], list);
+                printAndSave(taskCount);
+                System.out.println("Nice! I've marked this task as done:");
+                break;
+            case "unmark":
+                MarkTask.unmark(tempList[1], list);
+                printAndSave(taskCount);
+                System.out.println("OK, I've marked this task as not done yet:");
+                break;
+            case "delete":
+                System.out.println("Noted. I've removed this task:");
+                DeleteTask.delete(tempList[1], list);
+                printAndSave(taskCount - 1);
+                break;
+        }
+        System.out.println("------------------------------------------------------------");
+    }
+
+    /**
+     * Shows the number of task being track.
+     */
+    private static void printAndSave(int taskCount) {
+        if (taskCount + 1 == 1) {
```

Comment:
Not sure if you need to keep track of the `taskCount` because Java `List` should already do the job for you?